src/internal/api/authenticator_test.go

```diff
@@ -261,127 +277,78 @@ func TestLogout(t *testing.T) {
+		5*time.Second,
+		1*time.Second,
+		1*time.Second,
+		1,
+		1,
+	)
+
+	handler := func(w http.ResponseWriter, _ *http.Request) {
+		data := fmt.Sprintf(`{"accessToken":"%s","refreshToken":"%s"}`,
+			accessToken,
+			refreshToken,
+		)
+
+		w.WriteHeader(http.StatusNotFound)
+
+		_, _ = w.Write([]byte(data))
 	}
+	server := httptest.NewServer(http.HandlerFunc(handler))
 
-	for _, val := range testCases {
-		v := val
-		t.Run(v.name, func(t *testing.T) {
-			t.Parallel()
+	defer server.Close()
 
-			// mock cfgSvc
-			cfg := config.Config{Credentials: config.Credentials{}, Backoff: v.backoffCfg}
-			storage := mockedstorage.Storage[*config.Config]{}
-			storage.On("Model").Return(&cfg)
-			storage.On("Save").Return(v.saveError)
+	cfg := config.Config{Credentials: config.Credentials{
+		AccessToken:           accessToken,
+		RefreshToken:          refreshToken,
+		RefreshTokenExpiresAt: time.Now().Add(time.Hour),
+		AccessTokenExpiresAt:  time.Now(),
+	}, Backoff: config.BackoffCfg{Length: "0"}}
 
-			notificationManager := &NotificationMock{}
-			notificationManager.On("Event", &notification.Event{EventName: notificationEaseeStatusOffline}).Return(nil)
-
-			cfgDir := path.Join("./../../testdata/testing/", "")
-			mqttCfg := config.New(cfgDir)
-			v.auth.mqtt = fimpgo.NewMqttTransport(
-				mqttCfg.MQTTServerURI,
-				mqttCfg.MQTTClientIDPrefix,
-				mqttCfg.MQTTUsername,
-				mqttCfg.MQTTPassword,
-				true,
-				1,
-				1)
-
-			v.auth.notificationManager = notificationManager
-			v.auth.cfgSvc = config.NewService(&storage)
-			err := v.auth.handleFailedRefreshToken(v.errIn)
-			assert.Equal(t, v.expectedStatus, v.auth.status)
-			assert.Contains(t, err.Error(), v.errorContains)
-			notificationManager.AssertNumberOfCalls(t, "Event", v.notificationManagerCalled)
-		})
+	storage := mockedstorage.Storage[*config.Config]{}
+	storage.On("Model").Return(&cfg)
+	storage.On("Save").Return("ok")
+
+	cfgSrv := config.NewConfigServiceWithStorage(&storage)
+
+	notificationManager := &NotificationMock{}
+
+	httpClient := &http.Client{Timeout: 3 * time.Second}
+	client := NewHTTPClient(cfgSrv, httpClient, server.URL)
+	auth := authenticator{
+		http:                client,
+		cfgSvc:              config.NewService(&storage),
+		notificationManager: notificationManager,
+		backoff:             backoff,
+		mqtt: fimpgo.NewMqttTransport(
+			cfg.MQTTServerURI,
+			cfg.MQTTClientIDPrefix,
+			cfg.MQTTUsername,
+			cfg.MQTTPassword,
+			true,
+			1,
+			1,
+		),
 	}
-}
 
-func TestHookResetToReconnecting(t *testing.T) {
-	t.Parallel()
+	_, err := auth.AccessToken()
+	assert.Contains(t, err.Error(), "failed to perform token refresh api call:")
 
-	cfg := config.Config{Backoff: config.BackoffCfg{Length: "0"}}
-	storage := mockedstorage.Storage[*config.Config]{}
-	storage.On("Model").Return(&cfg)
+	for i := 0; i < 10; i++ {
+		_, err = auth.AccessToken()
+		assert.Contains(t, err.Error(), "too many requests, backoff is in use")
+	}
+
+	time.Sleep(5 * time.Second)
```

Comment:
As we spoke on a call - this was just an initial value to start writing the test. Please make this sleep as short as possible.